src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -300,41 +303,22 @@ double Camera::extract_near_z() const
     return near_z;
 }
 
-// Assumed shutter opening/closing curve is straight line.
-float Camera::integrate_inverse_curve(float a, float b, float x) const
-{
-    float constant;
-    int root_sign;
-    // opening
-    if(b == 0){
-        constant = 0;
-        root_sign = 1;
-    }
-    // closing
-    else{
-        constant = 1 - (a/2) - b;
-        root_sign = -1;
-    }
-    return root_sign * sqrt(2 * (x + (b * b) / (2 * a) - constant) / a) - b / a;
-}
-
-// Assumed shutter opening/closing curve is straight line.
-float Camera::map_sample(float sample) const
+float Camera::map_to_shutter_curve(float sample) const
 {
     // shutter is opening
-    if(0 <= sample && sample < m_height * m_normalized_open_end_time / 2)
+    if (0 <= sample && sample < m_height * m_normalized_open_end_time_half)
```

Comment:
Looks like the entire upper bound here can be precomputed.

Reply:
Oh, I worried about too many member variables(although I already made quite many member variables)..